core/templates/pages/exploration-editor-page/services/router.service.spec.ts

```diff
@@ -420,7 +409,29 @@ describe('Router Service', function() {
   it('should save pending changes', function() {
     var broadcastSpy = spyOn($rootScope, '$broadcast').and.callThrough();
     RouterService.savePendingChanges();
-    $timeout.flush();
     expect(broadcastSpy).toHaveBeenCalledWith('externalSave');
   });
+
+  it('should save pending changes even when AngularJS throws an error',
+    function() {
+      var broadcastSpy = spyOn($rootScope, '$broadcast');
```

Comment:
As discussed, let's try and get rid of the $destroy stuff here, and mock $broadcast so it does specific things on each call. (Please also add clear explanations for why we're writing it like this.) Thanks!

Reply:
Done! PTAL @seanlip